client-side/src/components/profileComponents/updateProfileCpmponent.jsx

```diff
@@ -0,0 +1,408 @@
+        try {
+            const updatedWebsites = await Promise.all(formData.websites.map(async (website) => {
+                if (!website.websiteId || website.websiteId === '') {
+                    const newWebsite = await createWebsite({ name: website.name, url: website.url });
+                    return {
+                        websiteId: newWebsite._id,
+                        name: website.name,
+                        url: website.url,
+                        status: website.status,
+                        limitedMinutes: website.limitedMinutes
+                    };
+                } else {
+                    const updatedWebsiteData = {
+                        name: website.name,
+                        url: website.url,
+                        status: website.status,
+                        limitedMinutes: website.limitedMinutes
+                    };
+                    await updateWebsite(website.websiteId, updatedWebsiteData);
+                    return {
+                        websiteId: website.websiteId,
+                        ...updatedWebsiteData
+                    };
+                }
+            }));
+
+            const updatedProfile = {
+                userId: formData.userId,
+                profileName: formData.profileName,
+                statusBlockedSites: formData.statusBlockedSites,
+                listWebsites: updatedWebsites,
+                timeProfile: {
+                    start: formData.timeProfile.timeStart,
+                    end: formData.timeProfile.timeEnd
+                },
+            };
+            await updateProfileApi(profile._id, updatedProfile);
+            setToastOpen(true);
+            setToastType('success');
+            setToastMessage('Profile updated successfully!');
+            handleClose();
+            if (onProfileUpdated) onProfileUpdated({ id: profile._id, ...updatedProfile });
+        } catch (error) {
+            console.error('Error updating profile:', error);
+            setToastOpen(true);
+            setToastType('error');
+            setToastMessage('Error updating profile!');
+        }
+    };
+
+    const handleDelete = async () => {
+        if (profile && profile._id) {
+            try {
+                await deleteProfileApi(profile._id);
+                dispatch(deleteProfile(profile._id));
+                setToastOpen(true);
+                setToastType('error');
+                setToastMessage('Profile deleted successfully!');
+            } catch (err) {
+                console.error('Error handling delete:', err);
+                setToastOpen(true);
+                setToastType('error');
+                setToastMessage('Error deleting profile!');
+            }
+        }
+    };
+
+    const isFormValid = () => {
+        return formData.profileName.trim() !== '' &&
+            formData.timeProfile.timeStart.trim() !== '' &&
+            formData.timeProfile.timeEnd.trim() !== '';
+    };
+
```

Comment:
the component is too long, please move logic to profileUtil.js
as well, check which of your functions should be memoized with useCallback()